Bottlenose/bottlenose.cpp

```diff
@@ -0,0 +1,132 @@
+#include <bottlenose.h>
+//DEFAULTS
+int _currentPin = 8;
+int _msDot = 50;
+int _msDash = 350;
+int _msCharGap = 100;
+int _highDelay = 2;
+int _lowDelay = 3;
+
+int _digits[10][5] ={
+    {1,1,1,1,1}, // 0
+    {0,1,1,1,1}, // 1
+    {0,0,1,1,1}, // 2
+    {0,0,0,1,1}, // 3
+    {0,0,0,0,1}, // 4
+    {0,0,0,0,0}, // 5
+    {1,0,0,0,0}, // 6
+    {1,1,0,0,0}, // 7
+    {1,1,1,0,0}, // 8
+    {1,1,1,1,0}, // 9
+};
+//CONSTRUCTOR
+bottlenose::bottlenose(void)
+{
+  //ADD STUFF
+}
+//PUBLIC
+bool bottlenose::begin(int pin)
+{
+  _currentPin = pin;
+  pinMode(_currentPin,OUTPUT);
+  return true;
+}
+bool bottlenose::writeDigit(int digit)
+{
+  runNumber(_digits[digit]);
+  return true;
+}
+bool bottlenose::writeChar(char character)
+{
+  //TODO Alphabet
+  return 0;
+}
+bool bottlenose::writeString(String character)
+{
+  //TODO String implementation
+  return 0;
+}
+int bottlenose::map(long sensorVal,long sensorLowerBound,long sensorUpperBound,long delayLowerBound,long delayUpperBound){
+  if(sensorVal < sensorLowerBound) sensorVal = sensorLowerBound;
+  if(sensorVal > sensorUpperBound) sensorVal = sensorUpperBound;
+  int thisDelay =  (sensorVal - sensorLowerBound) * (delayUpperBound - delayLowerBound) / (sensorUpperBound - sensorLowerBound) + delayLowerBound;
+  digitalWrite(_currentPin,HIGH);
+  delay(5);
+  digitalWrite(_currentPin,LOW);
+  delay(thisDelay);
+  return thisDelay;
+}
+//CONFIGS N SUCH
+//SETTERS
+int bottlenose::setDotLength(long ms)
```

Comment:
Should we also return an error code in functions like this so the client knows if they're putting invalid arguments in? Like a struct{ int result; int error}